feature/experimental/policy/policy_vrf_selection/base_vrf_selection/base_vrf_selection_test.go

```diff
@@ -0,0 +1,464 @@
+}
+
+func configInterfaceDUT(i *telemetry.Interface, me, peer *attrs.Attributes, subintfindex uint32, vlan uint16) *telemetry.Interface {
+	i.Description = ygot.String(me.Desc)
+	i.Type = telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	//Create subinterface
+	s := i.GetOrCreateSubinterface(subintfindex)
+
+	if vlan != 0 {
+		//Add VLANs
+		singletag := s.GetOrCreateVlan().GetOrCreateMatch().GetOrCreateSingleTagged()
+		singletag.VlanId = ygot.Uint16(vlan)
+	}
+	//Add IPv4 stack
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	a := s4.GetOrCreateAddress(me.IPv4)
+	a.PrefixLength = ygot.Uint8(plen4)
+
+	//Add IPv6 stack
+	s6 := s.GetOrCreateIpv6()
+	if *deviations.InterfaceEnabled {
+		s6.Enabled = ygot.Bool(true)
+	}
+	s6.GetOrCreateAddress(me.IPv6).PrefixLength = ygot.Uint8(plen6)
+
+	return i
+}
+
+// Configure Network instance on the DUT
+func configNetworkInstance(name string, peer *attrs.Attributes) *telemetry.NetworkInstance {
+
+	d := &telemetry.Device{}
+	ni := d.GetOrCreateNetworkInstance(name)
+
+	ni.Type = telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L2L3
+	static := ni.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "STATIC")
+	ipv4Nh := static.GetOrCreateStatic("0.0.0.0/0").GetOrCreateNextHop(peer.IPv4)
+	ipv6Nh := static.GetOrCreateStatic("::/0").GetOrCreateNextHop(peer.IPv6)
+	ipv4Nh.Recurse = ygot.Bool(true)
+	ipv6Nh.Recurse = ygot.Bool(true)
+
+	return ni
+}
+
+func configForwardingPolicy(name string, matchType string, ate *attrs.Attributes) *telemetry.NetworkInstance_PolicyForwarding {
+
+	d := &telemetry.Device{}
+	ni := d.GetOrCreateNetworkInstance(name)
+	ipv4Address := "0.0.0.0/0"
+	ipv6Address := "::/0"
+	//No support dscpset := []uint8{42, 46}
+
+	//Match policy
+	policyFwding := ni.GetOrCreatePolicyForwarding()
+
+	fwdPolicy1 := policyFwding.GetOrCreatePolicy("Match-IpipDscp4246")
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv4().Dscp = ygot.Uint8(46)
+	//No support fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv4().DscpSet = dscpset
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv4().Protocol = telemetry.PacketMatchTypes_IP_PROTOCOL_IP_IN_IP
+	fwdPolicy1.GetOrCreateRule(2).GetOrCreateIpv4().Dscp = ygot.Uint8(42)
+	fwdPolicy1.GetOrCreateRule(2).GetOrCreateIpv4().Protocol = telemetry.PacketMatchTypes_IP_PROTOCOL_IP_IN_IP
+	fwdPolicy1.GetOrCreateRule(2).GetOrCreateAction().NetworkInstance = ygot.String("10")
+	fwdPolicy1.GetOrCreateRule(3).GetOrCreateIpv4().SourceAddress = ygot.String(ipv4Address)
+	fwdPolicy1.GetOrCreateRule(3).GetOrCreateAction().Discard = ygot.Bool(true)
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv6().Dscp = ygot.Uint8(46)
+	//No support fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv6().DscpSet = dscpset
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv6().Protocol = telemetry.PacketMatchTypes_IP_PROTOCOL_IP_IN_IP
```

Comment:
Is V4 over V6 a requirement for Feature Profile ? @dplore , @xw-g

Reply:
If you are referring  to this code, its not for v4 over v6 but v4 and v6 rules taken care separately in a single policy.

Reply:
For this line #221, you are matching protocol IPinIP in IPv6 rule, which means you want to match outer header as IPv6 and inner header as IPv4(protcol-id/next-header 4)